lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -621,6 +637,34 @@ def _split(self, node):
 
         return before, after
 
+    def sample_from_channel(self, channel_slice, include_dead_time=True) -> Slice:
+        """Sample channel data using the time points corresponding to this track
+
+        For each time point in the track, sample the channel data corresponding to that kymograph
+        scan line.
+
+        .. note::
+
+            Tracks may skip kymograph frames when the signal drops below a certain level. If you
+            intend to sample every time point on the track, remember to interpolate the track
+            first using :meth:`KymoTrack.interpolate()`
+
+        Parameters
+        ----------
+        channel_slice : Slice
+            Sample from this channel data.
+        include_dead_time : bool
+            Include the time that the mirror returns to its start position after each kymograph
+            line.
+        """
+        ts_ranges = self._kymo.line_timestamp_ranges(include_dead_time=include_dead_time)
+        try:
+            return channel_slice.downsampled_over(
+                [ts_ranges[time_idx] for time_idx in self.time_idx]
```

Comment:
How is it guaranteed that each available time index also exists in `ts_ranges`, especially if `include_dead_time` is `False`? Wouldn't `ts_ranges` then miss a couple of indices that exist in `self.time_idx`?

Reply:
Why would `include_dead_time` matter? Include dead time impacts the duration of each timestamp range interval, not the number of ranges that are there.

Reply:
Don't you get a different set of indices, depending on that flag?